### PR TITLE
Add github actions

### PR DESCRIPTION
**Pull request by louisjdmartin**
## Proposed changes

> _Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue._

Add github actions to this project

## Types of changes

What types of changes does your code introduce to this plugin?
> _Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Issues closed by changes

> _List here all issues closed by your changes. Use a list of items like `- [x] Close #0`_

## Checklist

> _Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/lequal/sonar-icode-cnes-plugin/blob/master/CONTRIBUTING.md) doc
- [x] I agree with the [CODE OF CONDUCT](https://github.com/lequal/sonar-icode-cnes-plugin/blob/master/CONTRIBUTING.md)
- [x] Lint and unit tests pass locally with my changes
- [x] SonarCloud and Github Actions tests pass with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ x Any dependent changes have been merged and published in downstream modules

## Further comments

> If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: .github/workflows/java-continuous-integration.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        with:
+          distribution: 'adopt'
+          java-version: 11
+      - name: Cache Maven packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+      - name: Cache node_modules
```
Comment (Sancretor) [inline .github/workflows/java-continuous-integration.yml:33]: Are your sure you need node_modules here ? I don't remember I-Code was adding anything on the front side...

File: .github/workflows/java-continuous-integration.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        uses: actions/cache@v2
+        with:
+          path: node_modules
```
Comment (Sancretor) [inline .github/workflows/java-continuous-integration.yml:66]: Same here

File: .github/workflows/java-continuous-integration.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+      - name: Build and analyze
+        env:
+          # Needed to get some information about the pull request, if any
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          # SonarCloud access token should be generated from https://sonarcloud.io/account/security/
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
```
Comment (Sancretor) [inline .github/workflows/java-continuous-integration.yml:80]: Did you make sure those variables are declared for this project ?
To be honest, I don't remember where they are...

File: .github/workflows/java-continuous-integration.yml [hunk 4]
```diff
@@ -0,0 +1,81 @@
+# Description
+# ===========
+# This workflow is triggered each time
+# commits are pushed to GitHub or a pull request is opened.
+# It launches three jobs in parallel : a build with java 8,
+# a build with java 11 and a SonarCloud analysis.
+---
+name: Java CI
+
+on: [push, pull_request]
```
Comment (Sancretor) [inline .github/workflows/java-continuous-integration.yml:10]: I don't know if you saw it but there is also a file to handle automatic release + draft release notes.
Maybe we could add it too, don't you think ?

File: .github/workflows/java-continuous-integration.yml (hunk 3 above)
Comment (louisjdmartin) [reply .github/workflows/java-continuous-integration.yml:80]: Variables exists at organization level, you can find values here: https://github.com/cnescatlab/sonar-icode-cnes-plugin/settings/secrets/actions

File: .github/workflows/java-continuous-integration.yml (hunk 1 above)
Comment (louisjdmartin) [reply .github/workflows/java-continuous-integration.yml:33]: I think I was pretty generous on copy/paste, I'm going to remove this

File: .github/workflows/java-continuous-integration.yml (hunk 4 above)
Comment (louisjdmartin) [reply .github/workflows/java-continuous-integration.yml:10]: I can add this functionality too, I can check what is already done. 
It's used by cnes-report, no ?

Comment (Sancretor) [reply .github/workflows/java-continuous-integration.yml:10]: yes it is

**Review by Sancretor — APPROVED** [review]

**Pull request merged**